source/particle/generator/uniform_box.cc

```diff
@@ -79,8 +80,11 @@ namespace aspect
                       // worry about it and move on to next point.
                       try
                         {
-                          particles.insert(this->generate_particle(particle_position,particle_index));
+                          // Moved particle_index++ ABOVE the call to particles.insert in
+                          // order to ensure that all particles generated on all processers
+                          // have a unique particle id (i.e., particle_index)
```

Comment:
Let's remove this comment. When reading code, what matters only how it looks *now*, not through which process we *arrived* here.